mozilla_django_oidc/auth.py

```diff
@@ -105,22 +115,19 @@ def authenticate(self, code=None, state=None):
             user_response.raise_for_status()
             user_info = user_response.json()
             email = user_info.get('email')
-            if not email:
-                return None
 
-            create_user = False
-            try:
-                return self.UserModel.objects.get(email=email)
-            except self.UserModel.MultipleObjectsReturned:
+            # email based filtering
+            users = self.filter_users_by_claims(user_info)
+
+            if len(users) == 1:
```

Comment:
If ``users == None`` it raises an error because ``object of type 'NoneType' has no len()``